enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -10177,8 +10191,12 @@ class AdjointGenerator
                    backwardsShadow) ||
                   (Mode == DerivativeMode::ForwardModeSplit &&
                    backwardsShadow)) {
-                if (!inLoop)
-                  zeroKnownAllocation(bb, anti, args, *called, gutils->TLI);
+                if (!inLoop) {
+                  auto rule = [&](Value *anti) {
+                    zeroKnownAllocation(bb, anti, args, *called, gutils->TLI);
+                  };
+                  applyChainRule(bb, rule, anti);
+                }
```

Comment:
The `  !isa<AllocaInst>(anti)) {` part of the check below needs to be modified, since presumably in vector mode that won't be the case for the equivalent condition.

Reply:
done